messaging/src/main/java/org/wildfly/swarm/messaging/runtime/RemoteConnectionSocketBindingCustomizer.java

```diff
@@ -36,46 +35,31 @@
 
     @Inject
     @ConfigurationValue(MessagingProperties.REMOTE_MQ_NAME)
-    String mqName;
+    Optional<String> mqName = Optional.empty();
 
     @Inject
     @ConfigurationValue(MessagingProperties.REMOTE_HOST)
-    String remoteHost;
+    Optional<String> remoteHost = Optional.empty();
 
     @Inject
     @ConfigurationValue(MessagingProperties.REMOTE_PORT)
-    String remotePort;
+    Optional<String> remotePort = Optional.empty();
 
     @Override
     public void customize() {
         List<Server> servers = fraction.subresources().servers();
 
-        String mqName = this.mqName;
-        if (mqName == null) {
-            mqName = MessagingProperties.DEFAULT_REMOTE_MQ_NAME;
-        }
-
-        String finalMqName = mqName;
+        String mqName = this.mqName.orElse(MessagingProperties.DEFAULT_REMOTE_MQ_NAME);
 
         servers.stream()
                 .filter(e -> e instanceof EnhancedServer)
                 .forEach(server -> {
                     ((EnhancedServer) server).remoteConnections()
                             .forEach(connection -> {
                                 OutboundSocketBinding binding = new OutboundSocketBinding(connection.name());
-                                if (connection.name().equals(finalMqName)) {
-                                    String host = this.remoteHost;
-                                    if (host == null) {
-                                        host = connection.host();
-                                    }
-
-                                    String port = this.remotePort;
-                                    if (port == null) {
-                                        port = connection.port();
-                                    }
-
-                                    binding.remoteHost(host)
-                                            .remotePort(port);
+                                if (connection.name().equals(mqName)) {
+                                    binding.remoteHost(this.remoteHost.orElse(connection.host()))
```

Comment:
This could be written as `binding.remoteHost(this.remoteHost.orElseGet(connection::host))` to avoid calling `connection.host()` but I don't think it really affects performance in this case